### ISSUE TITLE
Typings for IPFS are broken

### ISSUE THREAD

**Issue by pleerock** [ipfs/js-ipfs#4232]
If you just do

```ts
import { IPFS } from "ipfs-core"

const ipfs: IPFS
ipfs.stop()
```

It will give you an error ` Property 'stop' does not exist on type 'IPFS<{}>'.`.

You have this error because types are simply broken. You have an IPFS class declared in the typings, but you also have a type IPFS declared (and seem to override) original IPFS class.

**Comment by achingbrain** [comment]
The `IPFS` type that is exported represents the [Core API](https://github.com/ipfs/js-ipfs/tree/master/docs/core-api) rather than `ipfs-core` (granted they are tediously similarly named - names are hard!).

The Core API is essentially what is exposed by the [HTTP RPC API](https://docs.ipfs.tech/reference/kubo/rpc/), which notably doesn't have a method to shut the node down, so the error message `Property 'stop' does not exist on type 'IPFS<{}>'.` is correct.

Do you see the error when you let tsc derive the type from the `create` method?

```ts
import { create } from 'ipfs-core'

const node = await create()
await node.stop()
```

**Comment by pleerock** [comment]
> Do you see the error when you let tsc derive the type from the create method?

nope, but that's isn't my use case. Imagine a following use case:

```ts
// some other file
export async function doSomethingForExampleStopMe(node: IPFS) {
   await node.stop()
}

// main file
const node = await create()
await doSomethingForExampleStopMe(node)
```

You'll have an error here because of type mismatch. And there is no exported type in the package I can use in order to fix the issue. (there is a way how I can do it basically by getting a returning type of the create function, but that's something common and should go out of the box).

**Comment by pleerock** [comment]
ohh these bots... @github-actions I already posted more details above ^

**Comment by whizzzkid** [comment]
js-ipfs is being deprecated in favor of [Helia](https://github.com/ipfs/helia). You can https://github.com/ipfs/js-ipfs/issues/4336 and [read the migration guide](https://github.com/ipfs/helia/wiki/%F0%9F%9A%9B-Migrating-from-js-IPFS).


I believe this should be resolved in Helia, if not, please let us know by reopening this issue before 2023-06-05!
